insights/parsers/pam.py

```diff
@@ -288,6 +312,39 @@ def __len__(self):
         """(int): Return the number of entries read from the file"""
         return len(self.data)
 
+    def search(self, **kwargs):
+        """
+        Search the pam.d configuration file by keyword.  This is provided by
+        the :func:`insights.parsers.keyword_search` function - see its
+        documentation for more information.
+
+        Searching on the list of PAM configuration entries is exactly like
+        they were dictionaries instead of objects with properties.  In
+        addition, the 'control_flags' property becomes a dictionary of
+        keywords and values, so that 'control_flags__contains' allows
+        searching for a particular control flag.
+
+        Returns:
+            (list): A list of PamConfEntry objects that match the given
+            search criteria.
+        """
+        # Because keyword_search takes dicts, and we have objects with complex
+        # properties, we convert them to a list of dicts.
+        # First, store the things we're going to convert.
+        prop_keys = ('service', 'interface', 'module_name', 'module_args')
+        search = []
+        # Can't find a neat way to do this as a comprehension, so it's back
+        # to loops.
+        for obj in self.data:
+            row = {'entry_obj': obj}
+            for key in prop_keys:
+                row[key] = getattr(obj, key)
+            # Convert control_flags down to key/value dictionary
+            row['control_flags'] = {cf.flag: cf.value for cf in obj.control_flags}
```

Comment:
@PaulWay this dict comprehension will need to be changed to be compatible with Python 2.6.  One solution would be to instead use:
```
row['control_flags'] = dict((cf.flag, cf.value) for cf in obj.control_flags)
```

Reply:
OK, will do.